packages/cli/src/commands/init/init.js

```diff
@@ -155,9 +165,20 @@ async function createFromReactNativeTemplate(
   }
 }
 
-function createProject(projectName: string, options: Options, version: string) {
-  fs.mkdirSync(projectName);
-  process.chdir(projectName);
+function createProject(
+  projectPath: string,
+  projectName: string,
+  options: Options,
+  version: string,
+) {
+  try {
+    fs.mkdirSync(projectPath, {recursive: true});
+    process.chdir(projectPath);
+  } catch (err) {
+    throw new Error(
+      'Cannot initialize new project because custom project directory does not exist.',
```

Comment:
Hm, I don't think that should happen. `mkdirp` can throw but on lack of permissions. We should forward the error correctly, not mask it without stack trace.

Reply:
Yes, you can use CLIError for this use-case. It allows you to re-throw an error with a better message.